packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart

```diff
@@ -175,6 +175,7 @@ abstract class ImagePickerPlatform extends PlatformInterface {
   /// in this call. You can then call [getLostData] when your app relaunches to retrieve the lost data.
   ///
   /// If no images were picked, the return value is null.
+  @Deprecated('Use getImageFromSource instead.')
```

Comment:
`getImageFromSource` uses `getImage`, should we flip the default implementation so the deprecated method calls the recommended one internally?

```
  Future<XFile?> getImageFromSource({
    required ImageSource source,
    ImagePickerOptions options = const ImagePickerOptions(),
  }) {
    return getImage(
      source: source,
      maxHeight: options.maxHeight,
      maxWidth: options.maxWidth,
      imageQuality: options.imageQuality,
      preferredCameraDevice: options.preferredCameraDevice,
    );
  }
```

Reply:
> `getImageFromSource` uses `getImage` in this file:

That's the default, not very good implementation, which platform implementations should override with a real one. See the paragraph starting "Note:" in https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-platform-interface-method-parameters

> Should we flip the default implementation so the deprecated method calls the recommended one internally?

That would suddenly make all old implementations throw when used with newer versions of the app-facing package.

Reply:
> The delegation of this method will feel backwards, because the implementing the new method in terms of the old method will discard the information from the new parameters. This is correct though, because the goal is to allow the app-facing package to call the new method regardless of whether implementation packages have implemented it.

Yep, exactly.

I'll flip the web implementation so the deprecated methods are based off of the new ones, so we can eventually remove the deprecated methods later!